### DISCUSSION TITLE
Building MSI for AppX (microsoft installer)

### DISCUSSION THREAD

**Discussion by uxduck** [marcelotduarte/cx_Freeze#1721] [Q&A] [answered]
I noticed that **msilib** is deprecated in Python 3.11, since microsoft is moving to Appx:
> Deprecated since version 3.11, will be removed in version 3.13: The [msilib](https://docs.python.org/3/library/msilib.html#module-msilib) module is deprecated (see [PEP 594](https://peps.python.org/pep-0594/#msilib) for details). [source](https://docs.python.org/3/library/msilib.html)

> Microsoft is slowly moving away from MSI in favor of Windows 10 Apps (**AppX**) as a new deployment model [source](https://peps.python.org/pep-0594/#msilib)

Did you have any thoughts if cx_Freeze should follow those changes?

---

Thanks for this project! It's been interesting to use it so far.

**Comment by marcelotduarte** [comment, marked as answer]
Not yet. But the options for msilib are:
1. Port to an independent package (See #2837); 
2. Port as internal package of cx_Freeze (like I did with bdist_msi).
3. Disable it for 3.13

Independently of the above options, a bdist_appx should be analyzed.

**Reply by daniilS** [reply]
@marcelotduarte with 3.13 expected to come out next month, I was wondering if any one of the options was settled on? I use cx_Freeze to distribute .msi installers, so would like to know if it will still be possible with Python 3.13, or if I should start looking for another tool to convert the frozen .exe and resources into an installer.

**Reply by marcelotduarte** [reply]
@daniilS I hope to release a version at the end of the month with support for Python 3.13, but for now it will be with bdist_msi disabled (3). But I will open an issue to see what support I will have to implement a new msilib (1) package, which I see will be feasible from a technical point of view.

**Reply by BobBicknell** [reply]
Do you have any update on when a version compatible with Python3.13 might be released

**Reply by marcelotduarte** [reply]
The version that supports Py 3.13 is ready, it just depends on a little more time for me to be able to make some adjustments and some tests with the samples we have.

**Reply by kristofmulier** [reply]
Hi @marcelotduarte ,
We look forward to try the new cx_freeze with Python 3.13

**Reply by nhaugrud** [reply]
Hi @marcelotduarte , is there an ETA on the version compatible with Python 3.13?

**Reply by marcelotduarte** [reply]
[Release 8.0](https://github.com/marcelotduarte/cx_Freeze/releases/tag/8.0.0) is out!
[Documentation](https://cx-freeze.readthedocs.io)
